tokio-stream/src/wrappers/watch.rs

```diff
@@ -40,6 +41,31 @@ use tokio::sync::watch::error::RecvError;
 /// let (tx, rx) = watch::channel("hello");
 /// let mut rx = WatchStream::new(rx);
 ///
+/// // existing rx output with "hello" is ignored here
+///
+/// tx.send("goodbye").unwrap();
+/// assert_eq!(rx.next().await, Some("goodbye"));
+/// # }
+/// ```
+///
+/// Example with [`WatchStream<T>::new_on_changed`]:
+///
+/// ```
+/// # #[tokio::main]
+/// # async fn main() {
+/// use futures::task::Poll;
+/// use futures_test::task::noop_context;
+/// use std::pin::Pin;
+/// use tokio::sync::watch;
+/// use tokio_stream::{Stream, StreamExt, wrappers::WatchStream};
+///
+/// let (tx, rx) = watch::channel("hello");
+/// let mut rx = WatchStream::new_on_changed(rx);
+///
+/// // no output from rx is available at this point
+/// let from_poll = Pin::new(&mut rx).poll_next(&mut noop_context());
+/// assert_eq!(from_poll, Poll::Pending);
```

Comment:
A little ugly; I think it would be ideal if we could add a utility function to help check whether a stream is pending or ready.